client/src/components/navigations/__tests__/ProfileModal.test.tsx

```diff
@@ -309,4 +315,55 @@ describe('[ProfileModal] rendering test', () => {
     expect(overlay.props.style.width).toBe(300);
     expect(overlay.props.style.height).toBe(252);
   });
+
+  it('Should be open when modalState.isMyself true', async () => {
+    const mockEnvironment = createMockEnvironment();
+
+    mockEnvironment.mock.queueOperationResolver((operation) =>
+      MockPayloadGenerator.generate(operation, {
+        User: (_, generateId) => generateUser(generateId(), false),
+      }),
+    );
+
+    const consumerRef = createRef<ConsumerRef>();
+
+    const screen = render(
+      <TestComponent consumerRef={consumerRef} environment={mockEnvironment} />,
+    );
+
+    act(() => consumerRef.current?.showModal({isMyself: true}));
+
+    expect(screen.queryAllByTestId('touch-add-friend')).toStrictEqual([]);
+    expect(screen.queryAllByTestId('touch-done')).toStrictEqual([]);
+    expect(screen.getByText('Chat with myself')).toBeTruthy();
```

Comment:
I think this would be better if you can get from our localized string.

```
getString('SELF_CHAT');
```

When `HackaTalk` becomes very popular we'd have to test all the localized strings 😄